### PR TITLE
TOSA specification in Arm partitioner

### PR DESCRIPTION
**Pull request by per**
Adds information on which TOSA specification is being used for the partitioner.

### CODE REVIEW

**Comment by per** [comment]
MacOS failure since commit 2f6d64f, not related to this change.

**Review by freddan80 — APPROVED** [review]

File: backends/arm/operator_support/tosa_supported_operators.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Checks if the fx.Node node is lowerable using the TOSA specification defined by tosa_spec.
+        To be implemented by subclasses targeting
+        """
+        raise NotImplementedError("NodeVisitor must be extended.")
+
+
+# container for all SupportedTosaOperatorCheck classes
+_tosa_spec_dicts: dict[TosaSpecification, dict[str, SupportedTOSAOperatorCheck]] = {
+    TosaSpecification.create_from_string("TOSA-0.80.0+BI"): {},
```
Comment (digantdesai) [inline backends/arm/operator_support/tosa_supported_operators.py:35]: Nit: Add a URL as a comment on how to find the exact version spec

File: backends/arm/operator_support/tosa_supported_operators.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+def get_registered_tosa_support_checks(
+    tosa_spec: TosaSpecification,
+) -> dict[str, SupportedTOSAOperatorCheck]:
+
+    if tosa_spec not in _tosa_spec_dicts:
+        raise RuntimeError
+
+    tosa_support_checks = {}
+    for target, tosa_check in _tosa_spec_dicts[tosa_spec].items():
+        tosa_support_checks[target] = tosa_check()
```
Comment (lucylq) [inline backends/arm/operator_support/tosa_supported_operators.py:61]: I think this should be:
```
tosa_support_checks[target] = tosa_check()
```
--> 
```
tosa_support_checks[target] = tosa_check
```

Comment (lucylq) [reply backends/arm/operator_support/tosa_supported_operators.py:61]: ah, wasn't this, but a type-check. resolved in https://github.com/pytorch/executorch/pull/6918

**Pull request merged**